src/helpers.rs

```diff
@@ -512,6 +516,39 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let value_place = op_place.offset(offset, MemPlaceMeta::None, layout, this)?;
         this.write_scalar(value, value_place.into())
     }
+
+    /// Parse a `timespec` struct and return it as a `std::time::Duration`. The outer `Result` is
+    /// for interpreter errors encountered while reading memory, and the inner `Result` indicates
+    /// whether the value in the `timespec` struct is invalid. Some libc functions will return
+    /// `EINVAL` if the struct's value is invalid.
+    fn read_timespec(
```

Comment:
IMO `src/shims/time.rs` would be a better place for this function. `helpers.rs` is only for things that are used by multiple shims.

Reply:
Oh, it *is* used by multiple shims. Never mind then. :)